### DISCUSSION TITLE
Joining output with comma

### DISCUSSION THREAD

**Discussion by basigu** [microsoft/RulesEngine#362] [Q&A]
I have an array of objects and object as a property 'Arrangement' and I would like to return all the unique values of this property.  What is the best way to do this.  Currently, I have the following code which is  not working.  
"Actions": {
    "OnSuccess": {
      "Name": "OutputExpression",
      "Context": {
        "Expression": "string.Join(',', input1Select(Arrangement).Distinct())"
      }
    }
  }

**Comment by abbasc52** [comment]
I see a c# expression issue here . string.Join accepts string as first parameter. So the expression should be  `string.Join(\",\", input1Select(Arrangement).Distinct())`

Also based on the type of Arrangement, you might want to do `.toString()` to ensure it has correct type.
e.g. `string.Join(\",\", input1Select(Arrangement.toString()).Distinct())`

Here is a working example for the above scenario -  https://dotnetfiddle.net/hUTRf1

**Reply by basigu** [reply]
Thank you abbasc52!
